### PR TITLE
Build with 1.58.1

### PR DESCRIPTION
**Pull request by haircommander**
#### What type of PR is this?

/kind dependency-change
#### What this PR does / why we need it:
as we need it to build on RHEL 8
#### Does this PR introduce a user-facing change?

```release-note
Downgrade packages to support MSRV 1.58.0
```

### CODE REVIEW

File: conmon-rs/server/Cargo.toml [hunk 1]
```diff
@@ -25,7 +25,7 @@ tempfile = "3.3.0"
 sendfd = { version = "0.4.3", features = ["tokio"] }
 prctl = "1.0.0"
 strum = { version = "0.24.1", features = ["derive"] }
-shadow-rs = "0.16.3"
+shadow-rs = { version = "0.16.3", features = [], default-features = false }
```
Comment (saschagrunert) [inline conmon-rs/server/Cargo.toml:28]:
```suggestion
shadow-rs = "0.16.1"
```

File: conmon-rs/server/Cargo.toml [hunk 2]
```diff
@@ -39,8 +39,8 @@ tz-rs = "0.6.14"
 tokio-fd = "0.3.0"
 
 [build-dependencies]
-shadow-rs = "0.16.3"
+shadow-rs = { version = "0.16.3", features = [], default-features = false }
```
Comment (saschagrunert) [inline conmon-rs/server/Cargo.toml:42]:
```suggestion
shadow-rs = "0.16.1"
```

File: Cargo.toml [hunk 3]
```diff
@@ -9,6 +9,5 @@ members = [
 debug = 2
 lto = true
 opt-level = "z"
-strip = true
```
Comment (saschagrunert) [inline Cargo.toml:12]: nit: let's strip the static binary:

```diff
diff --git a/Makefile b/Makefile
index d50e52c..a06bcf7 100644
--- a/Makefile
+++ b/Makefile
@@ -20,6 +20,7 @@ release:
 .PHONY: release-static
 release-static:
 	RUSTFLAGS="-C target-feature=+crt-static" cargo build --release --target x86_64-unknown-linux-gnu
+	strip -s target/x86_64-unknown-linux-gnu/release/conmonrs
 	ldd target/x86_64-unknown-linux-gnu/release/conmonrs | grep -q "statically linked"
 
 lint: lint-rust lint-go
```

**Comment by haircommander** [comment]
hm do you have any insight @saschagrunert https://github.com/containers/conmon-rs/actions/runs/3068048702/jobs/4955032084

**Comment by saschagrunert** [comment]
> hm do you have any insight @saschagrunert https://github.com/containers/conmon-rs/actions/runs/3068048702/jobs/4955032084

Yes we just have to change the grep text since we exchanged compilers.

File: Makefile [hunk 4]
```diff
@@ -20,7 +20,9 @@ release:
 .PHONY: release-static
 release-static:
 	RUSTFLAGS="-C target-feature=+crt-static" cargo build --release --target x86_64-unknown-linux-gnu
-	ldd target/x86_64-unknown-linux-gnu/release/conmonrs | grep -q "statically linked"
+	strip -s target/x86_64-unknown-linux-gnu/release/conmonrs
+	ldd target/x86_64-unknown-linux-gnu/release/conmonrs# | grep -q "not a dynamic executable"
```
Comment (saschagrunert) [inline Makefile:24]: It may output on stderr

Comment (rphillips) [reply Makefile:24]: `grep -qE '(statically linked)|(not a dynamic executable)'` would allow us to check for both

**Pull request merged**